ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -28,21 +28,23 @@ interface State {
 }
 
 const workflowEntrypoint = '<default>';
+type ParamSelector = 'parameters' | 'workflowParameters';
 
 export class SubmitWorkflowPanel extends React.Component<Props, State> {
     constructor(props: any) {
         super(props);
         const defaultTemplate: Template = {
             name: workflowEntrypoint,
             inputs: {
-                parameters: this.props.workflowParameters
+                parameters: []
             }
         };
-        const state = {
+        const state: State = {
             entrypoint: workflowEntrypoint,
             entrypoints: this.props.templates.map(t => t.name),
             selectedTemplate: defaultTemplate,
-            parameters: this.props.workflowParameters || [],
+            parameters: [] as Parameter[],
+            workflowParameters: this.props.workflowParameters,
```

Comment:
do we need to make a copy of this object rather than reference if we're going to change it? Use `Object.assign` to copy an object

Reply:
Changed to a copy in https://github.com/argoproj/argo-workflows/pull/7809/commits/44da063ac2cb9965d9321d81a0f36276f7db733b.